### PR TITLE
Stabilise 'External Kotlin Target' APIs

### PR DESCRIPTION
**Pull request by sellmair**

### CODE REVIEW

File: libraries/tools/kotlin-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -26,6 +27,12 @@ kotlin.sourceSets.all {
     languageSettings.optIn("org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi")
 }
 
+apiValidation {
+    publicMarkers.add("org.jetbrains.kotlin.gradle.ExternalKotlinTargetApi")
```
Comment (AmrAfifiy) [inline libraries/tools/kotlin-gradle-plugin/build.gradle.kts:31]: are all APIs annotated with @ExternalKotlinTargetApi end up in kotlin-gradle-plugin-api artifact?

Comment (sellmair) [reply libraries/tools/kotlin-gradle-plugin/build.gradle.kts:31]: No, those apis will be part of the `kotlin-gradle-plugin` artifact as those APIs are required to call into internal APIs of the implementation. So: `kotlin-gradle-plugin-api` already has binary compatibility validation, `kotlin-gradle-plugin` has not, but APIs marked with this annotation will have to be kept stable!

**Review by AmrAfifiy — COMMENTED** [review]
Looks good to me, thanks for the detailed documentation!

**Comment by sellmair** [comment]
I am closing the Pull Request on GitHub, since feedback from Google is OK. 
Merging and further discussion will proceed on Space.

**Pull request closed without merging**